File: frontend/src/types/backgroundTask.ts
export interface BackgroundTask {
  id: number
  task_type: string
  title: string
  status: 'running' | 'success' | 'error'
  progress_data: Record<string, unknown>
  result: Record<string, unknown> | null
  error: string | null
  created_at: string
  updated_at: string
}

File: backend/app/schemas/background_task.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class BackgroundTaskOut(BaseModel):
    id: int
    task_type: str
    title: str
    status: str
    progress_data: dict[str, Any]
    result: dict[str, Any] | None = None
    error: str | None = None
    created_at: datetime
    updated_at: datetime
